lib/ddtrace/contrib/presto/instrumentation.rb

```diff
@@ -0,0 +1,103 @@
+require 'ddtrace/pin'
+require 'ddtrace/ext/net'
+require 'ddtrace/ext/sql'
+require 'ddtrace/ext/app_types'
+require 'ddtrace/contrib/presto/ext'
+
+module Datadog
+  module Contrib
+    module Presto
+      # Instrumentation for Presto integration
+      module Instrumentation
+        # Instrumentation for Presto::Client::Client
+        module Client
+          def self.included(base)
+            base.send(:prepend, InstanceMethods)
+          end
+
+          # Instance methods for Presto::Client
+          module InstanceMethods
+            def run(query)
+              datadog_pin.tracer.trace(Ext::SPAN_QUERY) do |span|
```

Comment:
It doesn't seem like we need to use a `Pin` for this instrumentation: none of the information required for tracing needs to be persisted in the Presto `Client` instance class.

Could we change this to use a simple `tracer.trace` instead here? ([Here's an example](https://github.com/DataDog/dd-trace-rb/blob/bd853ede1f9d7c3984dcdea76b736fc0e0cb8d1f/lib/ddtrace/contrib/resque/resque_job.rb#L15))

Reply:
Thanks @marcotc for taking a look. I've updated this to use the simpler `tracer.trace` here, and to pass in `span_options` to simplify things a bit. Let me know what you think. Oddly there's some graphql specs failing now, though I don't believe this PR touches that code.